### PR TITLE
[MIG] dms_field: Migration to 14.0

### PR DESCRIPTION
**Pull request by olgamarcocb**
Standard Migration

### CODE REVIEW

**Comment by marylla** [comment]
@olgamarcocb Please check the code coverage. Can you improve it?

**Review by FernandoRomera — APPROVED** [review]
Local functional test, LGTM.

**Comment by olaf-wagner** [comment]
@olgamarcocb With the latest code version in 14.0, I had to replace `res_mimetype` by `mimetype` in `dms_field/static/src/js/base/dms_tree_controller.js`. With that change it seems to work fine for me locally. Perhaps you can update your branch and we try to get it merged?

**Comment by marcelsavegnago** [comment]
@olgamarcocb 

Considering the 14.0 migration script, I am submitting this PR. Please have a look and I hope I helped in some way.

https://github.com/tegin/dms/pull/3

https://github.com/OCA/dms/blob/46df04975e7df4976abf8003c25b8a0153674cd2/dms/migrations/14.0.1/pre-migration.py#L7

File: dms_field/security/ir.model.access.csv [hunk 1]
```diff
@@ -0,0 +1,2 @@
+id,name,model_id/id,group_id/id,perm_read,perm_write,perm_create,perm_unlink
+access_dms_add_directory_record,access_dms_add_directory_record,model_dms_add_directory_record,base.group_user,1,0,0,0
```
Comment (marcelsavegnago) [inline dms_field/security/ir.model.access.csv]: This way it gives an error when trying to add a storage to the record.

File: dms_field/static/src/js/base/dms_tree_controller.js [hunk 2]
```diff
@@ -0,0 +1,585 @@
...
+            return this._rpc({
+                model: "dms.file",
+                method: "search_read",
+                fields: _.union(args.file.fields || [], [
+                    "permission_read",
+                    "permission_create",
+                    "permission_write",
+                    "permission_unlink",
+                    "name",
+                    "res_mimetype",
```
Comment (marcelsavegnago) [inline dms_field/static/src/js/base/dms_tree_controller.js]: I believe the correct name now in 14.0 is mimetype.
https://github.com/OCA/dms/blob/46df04975e7df4976abf8003c25b8a0153674cd2/dms/migrations/14.0.1/pre-migration.py#L7

**Review by marcelsavegnago — CHANGES_REQUESTED** [review]

File: dms_field/static/src/js/base/dms_tree_controller.js (hunk 2 above)
Comment (olgamarcocb) [reply dms_field/static/src/js/base/dms_tree_controller.js]: Thank you!

**Comment by marcelsavegnago** [comment]
Hello @etobella 
Apparently the runboat was not available. Is it some REPO configuration?

**Comment by etobella** [comment]
I could access it without problem. 

Can you try?
https://runboat.odoo-community.org/webui/build.html?name=b83702787-0a42-482d-a5a8-b617b5a43ef3

**Comment by marcelsavegnago** [comment]
> I could access it without problem.
> 
> Can you try? https://runboat.odoo-community.org/webui/build.html?name=b83702787-0a42-482d-a5a8-b617b5a43ef3

Thanks. I try.

**Review by augustodinizl — APPROVED** [review]
Functional tests on runboat - OK
Screenshot at May 04 14-02-45
Screenshot at May 04 13-59-30

LGTM

Good work folks!

**Comment by marcelsavegnago** [comment]
Can you change the license from AGPL to LGPL?. This has already been done in version 13.0 and authorized by the module authors.
https://github.com/OCA/dms/pull/170

File: dms_field/static/src/js/base/dms_tree_controller.js [hunk 3]
```diff
@@ -0,0 +1,585 @@
...
+                }),
+            });
+            var dt = this._makeDataPoint({
+                data: data,
+                modelName: "dms.file",
+            });
+            return {
+                id: dt.id,
+                text: dt.data.display_name,
+                icon: mimetype.mimetype2fa(dt.data.res_mimetype, {prefix: "fa fa-"}),
```
Comment (marcelsavegnago) [inline dms_field/static/src/js/base/dms_tree_controller.js]:
```suggestion
                icon: mimetype.mimetype2fa(dt.data.mimetype, {prefix: "fa fa-"}),
```

File: dms_field/static/src/js/base/dms_tree_controller.js [hunk 4]
```diff
@@ -0,0 +1,585 @@
...
+                type: "file",
+                data: dt,
+            };
+        },
+        _onDMSPreviewFile: function (ev) {
+            var record = this._getDataPoint(ev.data.id, {raw: true});
+            var fieldName = "content";
+            var file_mimetype = record.data.res_mimetype;
```
Comment (marcelsavegnago) [inline dms_field/static/src/js/base/dms_tree_controller.js]:
```suggestion
            var file_mimetype = record.data.mimetype;
```

File: dms_field/static/src/js/base/dms_tree_controller.js [hunk 5]
```diff
@@ -0,0 +1,585 @@
...
+            if (
+                type === "video" ||
+                type === "image" ||
+                file_mimetype === "application/pdf"
+            ) {
+                var attachmentViewer = new FieldPreviewViewer(
+                    this,
+                    [
+                        {
+                            mimetype: record.data.res_mimetype,
```
Comment (marcelsavegnago) [inline dms_field/static/src/js/base/dms_tree_controller.js]:
```suggestion
                            mimetype: record.data.mimetype,
```

File: dms_field/static/src/js/base/dms_tree_controller.js [hunk 6]
```diff
@@ -0,0 +1,585 @@
...
+                            id: record.data.id,
+                            fileType: record.data.res_mimetype,
```
Comment (marcelsavegnago) [inline dms_field/static/src/js/base/dms_tree_controller.js]:
```suggestion
                            fileType: record.data.mimetype,
```

**Review by marcelsavegnago — CHANGES_REQUESTED** [review]

**Review by etobella — APPROVED** [review]

**Pull request merged**